pkg/registry/generic/rest/proxy.go

```diff
@@ -130,33 +137,35 @@ func (h *UpgradeAwareProxyHandler) tryUpgrade(w http.ResponseWriter, req *http.R
 	if !httpstream.IsUpgradeRequest(req) {
 		return false
 	}
-
-	backendConn, err := proxy.DialURL(h.Location, h.Transport)
-	if err != nil {
+	if err := h.handleUpgrade(w, req); err != nil {
```

Comment:
Good catch! It turns out this is an existing issue, since the ErrorResponder fails once the connection is hijacked. I added a test case for this, and fixed it by writing the errors to the hijacked connection directly.